shared/libraries/opcuatms/tests/opcuatms_integration/test_tms_integration.cpp

```diff
@@ -236,3 +237,84 @@ TEST_F(TmsIntegrationTest, GetDomainSignal)
     ASSERT_NO_THROW(domainSignal = byteStepSignal.getDomainSignal());
     ASSERT_TRUE(domainSignal.assigned());
 }
+
+TEST_F(TmsIntegrationTest, GetAvailableFunctionBlockTypes)
+{
+    InstancePtr device = createDevice();
+    TmsServer tmsServer(device);
+    tmsServer.start();
+
+    auto serverFbTypes = device.getAvailableFunctionBlockTypes();
+
+    TmsClient tmsClient(device.getContext(), nullptr, OPC_URL, nullptr);
+    auto clientDevice = tmsClient.connect();
+
+    const auto clientFbTypes = clientDevice.getAvailableFunctionBlockTypes();
+    
+    ASSERT_TRUE(TestComparators::FunctionBlockTypeDictEquals(serverFbTypes, clientFbTypes));
+}
+
+TEST_F(TmsIntegrationTest, AddFunctionBlock)
+{
+    InstancePtr device = createDevice();
+    TmsServer tmsServer(device);
+    tmsServer.start();
+
+    TmsClient tmsClient(device.getContext(), nullptr, OPC_URL, nullptr);
+    auto clientDevice = tmsClient.connect();
+
+    auto fb1 = clientDevice.addFunctionBlock("mock_fb_uid");
+    ASSERT_TRUE(fb1.assigned());
+    ASSERT_EQ("mock_fb_uid_1", fb1.getLocalId());
+
+    auto fb2 = clientDevice.addFunctionBlock("mock_fb_uid");
+    ASSERT_TRUE(fb2.assigned());
+    ASSERT_EQ("mock_fb_uid_2", fb2.getLocalId());
+
+    ASSERT_EQ(3, clientDevice.getFunctionBlocks().getCount());
+}
+
+TEST_F(TmsIntegrationTest, AddFunctionBlockWitchConfig)
+{
+    InstancePtr device = createDevice();
+    TmsServer tmsServer(device);
+    tmsServer.start();
+
+    TmsClient tmsClient(device.getContext(), nullptr, OPC_URL, nullptr);
+    auto clientDevice = tmsClient.connect();
+
+    const auto clientFbTypes = clientDevice.getAvailableFunctionBlockTypes();
+
+    auto config = clientFbTypes.get("mock_fb_uid").createDefaultConfig();
+    config.addProperty(IntProperty("TestConfigInt", 0)); // I realy dont like this, but there is no way of trasfering default config property object over opcua ...
```

Comment:
Any suggestions about this? Maybe we could allow DictObject as config parameter.